browser/caliban.py

```diff
@@ -473,6 +475,110 @@ def action_threshold(self, y1, x1, y2, x2, label):
+        predict_area = adjusted_raw_frame[y1:y2, x1:x2]
+        # import pdb; pdb.set_trace()
+
+        # returns 1 where label is predicted to be based on contouring, 0 background
+        contoured = morphological_chan_vese(predict_area, 100, init_level_set=level_set)
+
+        # contoured area should get original label value
+        contoured_label = np.where(contoured, label, 0)
+        # contours tend to fit very tightly, a small expansion here works well
+        contoured_label = dilation(contoured_label, disk(3))
+
+        # don't want to leave the original (un-contoured) label in the image
+        label_img = np.where(label_img == label, 0, label_img)
+
+        # never overwrite other labels with new contoured label
+        safe_overlay = np.where(label_img == 0, contoured_label, label_img)
+
+        # label must be present in safe_overlay for this to be a valid contour result
+        # very few pixels of contoured label indicate contour prediction not worth keeping
+        pixel_count = np.count_nonzero(safe_overlay == label)
+        if pixel_count < 20:
+            safe_overlay = np.copy(self.frame[y1:y2, x1:x2, self.feature])
+
+        # put it back in the full image so can use centroid coords for post-contour cleanup
+        full_frame = np.copy(self.frame[..., self.feature])
+        full_frame[y1:y2, x1:x2] = safe_overlay
+
+        # avoid automated label cleanup if the centroid (flood seed point) is not the right label
+        if full_frame[int(props['centroid'][0]), int(props['centroid'][1])] != label:
+            img_trimmed = full_frame
+        else:
+            # morphology and logic used by pixel-trimming action, with object centroid as seed
+            contig_cell = flood(image=full_frame,
+                                seed_point=(int(props['centroid'][0]), int(props['centroid'][1])))
+
+            # any pixels in img_ann that have value 'label' and are NOT connected to hole_fill_seed
+            # get changed to 0, all other pixels retain their original value
+            img_trimmed = np.where(np.logical_and(np.invert(contig_cell),
+                                                  full_frame == label),
+                                   0, full_frame)
+
+        # update image; cell_info should never change as a result of this
+        self.frame[y1:y2, x1:x2, self.feature] = img_trimmed[y1:y2, x1:x2]
+        self.y_changed = True
+
+    def action_erode(self, label):
+        """
+        Use morphological erosion to incrementally shrink the selected label.
+        """
+        img_ann = self.frame[..., self.feature]
+
+        # if label is adjacent to another label, don't let that interfere
+        img_erode = np.where(img_ann == label, label, 0)
+        # erode the label
+        img_erode = erosion(img_erode, square(3))
+
+        # put the label back in
+        img_ann = np.where(img_ann == label, img_erode, img_ann)
+
+        in_modified = np.any(np.isin(img_ann, label))
```

Comment:
If `label` is just an integer I think just checking equality is faster than `np.isin`.

```suggestion
        in_modified = np.any(img_ann == label)
```

Reply:
I investigated this with timeit and didn't find a difference. So I think it's down to the style you prefer here